enterprise/utils.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""
+Utilities to be used across the enterprise djangoapp
+"""
+try:
+    from openedx.core.djangoapps.api_admin.utils import course_discovery_api_client
+except ImportError:
+    course_discovery_api_client = None
+
+class NotConnectedToOpenEdX(Exception):
+    """
+    Exception that is raised when we try to import from
+    the core openedx repository, but it isn't available
+    """
+    pass
+
+
+def get_catalog_api_client(user):
+    """
+    Retrieve an authenticated API client that can be used to access the course
+    discovery API; raise an exception to be caught at a higher level if the
+    package doesn't have OpenEdX resources available.
+    """
+    if course_discovery_api_client is None:
+        raise NotConnectedToOpenEdX(
+            'To get a catalog API client, this package must be installed in an OpenEdX environment.'
+        )
+    else:
+        return course_discovery_api_client(user)
+
+
+def get_all_catalogs(user):
+    """
+    Return a list of all course catalogs, including name and ID
+    """
+    # Initialize the permanent list to hold all results, as well as
+    # the ephemeral list to store each API call's results.
+    results = []
+    this_result = []
```

Comment:
what is the difference b/w `results` and `this_result` ? , we should use some explanatory names.